README-DOCKER.md

```diff
@@ -0,0 +1,123 @@
+# mfem Docker
```

Comment:
I'm torn -- should we keep this here, or move to `config/docker` and mention it instead in `README` and/or `INSTALL`?

🤔

Reply:
The main issue to not have it in config/docker is mostly just the person that browses to the repo seeing there is a Dockerfile! I put it in the root because of that. The alternative that I think is a good design is to have a link in the README, but y'all don't have a standard markdown README so we can't do that. So if we are able to make the existence clear somewhere, I do like having it be like:

```
config/docker/README.md
```
because it will render nicely in GitHub! But if there is no way to make reference in the root I think we should keep it there.

Reply:
Lets do that... it is time to change the `README` to `README.md`

😱

Reply:
make it so!  
![image](https://user-images.githubusercontent.com/814322/166084083-f8948216-2350-40f9-b78c-d347431b381d.png)

And it is!
https://github.com/mfem/mfem/pull/2981/commits/8be4ab8d24ffe4e541adfa89a6fe00e760addd10

I preserved the nice emoji art in a comment block, here is how it looks rendered:

https://github.com/researchapps/mfem/tree/add/docker-build-for-demo